app/src/main/java/org/mozilla/fenix/perf/ProfilerUtils.kt

```diff
@@ -0,0 +1,150 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.perf
+
+import android.content.ClipData
+import android.content.ClipboardManager
+import android.content.Context
+import android.util.Base64
+import mozilla.components.concept.fetch.MutableHeaders
+import mozilla.components.concept.fetch.Request
+import mozilla.components.concept.fetch.Response
+import org.json.JSONObject
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.components
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+
+private const val PROFILER_API = "https://api.profiler.firefox.com/compressed-store"
+private const val PROFILER_SERVER_HEADER = "application/vnd.firefox-profiler+json;version=1.0"
+private const val TOKEN = "profileToken"
+private const val PROFILER_DATA_URL = "https://profiler.firefox.com/public/"
+
+private val firefox_features = arrayOf(
+    "screenshots", "js", "leaf", "stackwalk", "cpu", "java",
+    "processcpu", "ipcmessages", "java"
+)
+private val firefox_threads = arrayOf(
+    "GeckoMain", "Compositor", "Renderer",
+    "SwComposite", "DOM Worker"
+)
+
+private val graphics_features = arrayOf("java", "ipcmessages")
+private val graphics_threads = arrayOf(
+    "GeckoMain", "Compositor", "Renderer", "SwComposite",
+    "RenderBackend", "SceneBuilder", "WrWorker", "CanvasWorkers",
+)
+
+private val media_features = arrayOf(
+    "js", "leaf", "stackwalk", "cpu", "audiocallbacktracing",
+    "ipcmessages", "processcpu", "java"
+)
+private val media_threads = arrayOf(
+    "cubeb", "audio", "camera", "capture", "Compositor", "GeckoMain", "gmp", "graph", "grph",
+    "InotifyEventThread", "IPDL Background", "media", "ModuleProcessThread", "PacerThread",
+    "RemVidChild", "RenderBackend", "Renderer", "Socket Thread", "SwComposite",
+    "webrtc"
+)
+
+private val networking_features = arrayOf(
+    "screenshots", "js", "leaf", "stackwalk", "cpu", "java",
+    "processcpu", "ipcmessages"
+)
+
+private val networking_threads = arrayOf(
+    "Compositor", "DNS Resolver", "DOM Worker", "GeckoMain",
+    "Renderer", "Socket Thread", "StreamTrans", "SwComposite", "TRR Background"
+)
+
+/**
+ * Profiler settings enum for grouping features and settings together
+ */
+enum class ProfilerSettings(val threads: Array<String>, val features: Array<String>) {
+    Firefox(firefox_threads, firefox_features),
+    Graphics(graphics_threads, graphics_features),
+    Media(media_threads, media_features),
+    Networking(networking_threads, networking_features);
+}
+
+/**
+ * Collection of functionality to parse and save the profile returned by GeckoView.
+ */
+object ProfilerUtils {
+
+    private fun saveProfileUrlToClipboardAndToast(profileResult: ByteArray, context: Context): String {
```

Comment:
todo (blocking): this method no longer toasts so it shouldn't be named to suggest it does.